app/javascript/packs/prompt.ts

```diff
@@ -12,14 +21,41 @@ document.addEventListener("DOMContentLoaded", () => {
   document.body.appendChild(comp.$el);
 });
 
-function getPrompt() {
-  let prompts = document.querySelectorAll('.prompt-external-link');
-  let promptHash = new Map();
+const userPrompt = (() => {
+  const hasVisited = getCookie() !== null;
+
+  const getPrompt = () => {
+    let promptMap = new Map();
+    let prompt = document.querySelector('.prompt-external-link');
+  
+    promptMap.set('link', prompt ? prompt.attributes['data-link'].value : "");
+    promptMap.set('text', prompt ? prompt.attributes['data-text'].value : "");
+  
+    return promptMap
+  }
+
+  const promptOn = () => {
+    let prompt = getPrompt()
+    let linkEmpty = prompt.get('link') === ""
+    let textEmpty = prompt.get('text') === ""
 
-  prompts.forEach((p) => {
-    promptHash.set('link', p.attributes['data-link'].value);
-    promptHash.set('text', p.attributes['data-text'].value);
-  })
+    return !hasVisited && !linkEmpty && !textEmpty
+  }
 
-  return promptHash
-}
\ No newline at end of file
+  const setCookie = () => {
+    if (!hasVisited) {
+      let date = Date.now().toString()
+      localStorage.setItem('wd_visitedLocationShowPage', date)
```

Comment:
1. Let's make this key a constant, like `const visitedKey = 'visitedLocationShowPage'`, so that we can't accidentally misspell it between the two places we use it.
2. Since localstorage is scoped to the current website, let's drop the `wd_` prefix.